commit 9d42476f71a9a000ea2d72aae5f4d43c5061fe10
Author: Francois Dugast <francois.dugast@intel.com>
Date:   Wed Nov 13 17:22:12 2024 +0100

    drm/xe: Allow fault injection in vm create and vm bind IOCTLs
    
    Use fault injection infrastructure to allow specific functions to
    be configured over debugfs for failing during the execution of
    xe_vm_create_ioctl() and xe_vm_bind_ioctl(). This allows more
    thorough testing from user space by going through code paths for
    error handling and unwinding which cannot be reached by simply
    injecting errors in IOCTL arguments. This can help increase code
    robustness.
    
    v2: Add xe_pt_update_ops_{prepare,run} (Matthew Brost)
    
    Reviewed-by: Rodrigo Vivi <rodrigo.vivi@intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20241113162212.2154103-1-francois.dugast@intel.com
    Signed-off-by: Francois Dugast <francois.dugast@intel.com>

diff --git a/drivers/gpu/drm/xe/xe_exec_queue.c b/drivers/gpu/drm/xe/xe_exec_queue.c
index ff556773c106..aab9e561153d 100644
--- a/drivers/gpu/drm/xe/xe_exec_queue.c
+++ b/drivers/gpu/drm/xe/xe_exec_queue.c
@@ -240,6 +240,7 @@ struct xe_exec_queue *xe_exec_queue_create_bind(struct xe_device *xe,
 
 	return q;
 }
+ALLOW_ERROR_INJECTION(xe_exec_queue_create_bind, ERRNO);
 
 void xe_exec_queue_destroy(struct kref *ref)
 {